src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+        }
+    }
+
+    private void processBatchResponseFiles(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor, ArmBatchResponses armBatchResponses) {
+        armBatchResponses.getArmBatchResponseMap().values().forEach(
+            armResponseBatchData -> {
+                if (nonNull(armResponseBatchData.getInvalidLineFileFilenameProcessor())
+                    && (nonNull(armResponseBatchData.getCreateRecordFilenameProcessor())
+                    || nonNull(armResponseBatchData.getArmResponseUploadFileRecord()))) {
+
+                    beforeProcessingResponseFiles(armResponseBatchData.getExternalObjectDirectoryId());
+
+                    tryProcessInvalidLineFile(armResponseBatchData.getExternalObjectDirectoryId(),
+                                              armResponseBatchData.getInvalidLineFileFilenameProcessor(),
+                                              armResponseBatchData.getArmResponseInvalidLineRecord(),
+                                              armResponseBatchData.getCreateRecordFilenameProcessor(),
+                                              armResponseBatchData.getUploadFileFilenameProcessor());
+                    deleteResponseBlobs(armResponseBatchData);
+                } else if (nonNull(armResponseBatchData.getCreateRecordFilenameProcessor())
+                    && nonNull(armResponseBatchData.getUploadFileFilenameProcessor())) {
+
+                    beforeProcessingResponseFiles(armResponseBatchData.getExternalObjectDirectoryId());
+
+                    processUploadFileObject(batchUploadFileFilenameProcessor, armResponseBatchData);
+                    deleteResponseBlobs(armResponseBatchData);
+                } else {
+                    log.info("Unable to find response files for external object {}", armResponseBatchData.getExternalObjectDirectoryId());
+                    try {
+                        ExternalObjectDirectoryEntity externalObjectDirectory =
+                            getExternalObjectDirectoryEntity(armResponseBatchData.getExternalObjectDirectoryId());
+
+                        updateExternalObjectDirectoryStatus(externalObjectDirectory, EodHelper.armDropZoneStatus());
+
+                    } catch (Exception e) {
+                        log.error(UNABLE_TO_UPDATE_EOD, e);
+                    }
+                }
+            }
+        );
+    }
+
+    private void processCreateRecordResponseFiles(List<CreateRecordFilenameProcessor> createRecordResponses,
+                                                  ArmBatchResponses armBatchResponses) {
+        for (CreateRecordFilenameProcessor createRecordFilenameProcessor : createRecordResponses) {
+            try {
+                BinaryData createRecordBinary = armDataManagementApi.getBlobData(createRecordFilenameProcessor.getCreateRecordFilenameAndPath());
```

Comment:
Should we be using Binary data. If you look at the underlying call. ArmServiceImpl.getBlodData we use downloadContent().

Reply:
changes to existing code that are not necessary for plugging-in DETS functionality cannot be tackled in this PR, as explained in the PR description